docs/DafnyRef/UserGuide.md

```diff
@@ -124,12 +124,25 @@ The following commands are recognized. In each case all files listed are parsed
 - `dafny verify` -- verifies the listed files. Although the Dafny program being considered
 consists of the listed files and any included files (recursively), by default only listed files are verified.
 This is similar to using the option `/compile:0` in the old-style command-line. 
-- `dafny run` -- verifies, compiles and runs the Dafny program. The option `--no-verify` suppresses verification checks. The `-t` or `--target` option states the target platform to compile to (default is C#). 
-This is similar to the old-style options `/compile:3` or `/compile:4`.
 - `dafny translate` -- verifies the program and translates it to a source artifact, perhaps with information for
 a target language build tool, for the target platform, similar to the previous `/compile:0 '/spillTargetCode:2`.
 `--no-verify` suppresses verification checks.
 - `dafny build` -- verifies (except if `--no-verify` is used) and compiles the Dafny program, producing an executable artifact for the target system. The command has options that enable being specific about the build platform and architecture.
+- `dafny run` -- verifies, compiles and runs the Dafny program. The option `--no-verify` suppresses verification checks. The `-t` or `--target` option states the target platform to compile to (default is C#). 
+This is similar to the old-style options `/compile:3` or `/compile:4`.
+`dafny run` takes only one .dfy file. Any other .dfy file and any other
+command-line argument that is not an option or its value or is after a `--`
+argument is considered a command-line argument for the user program being 
+built and run. For example,
+  - `dafny run A.dfy` -- builds and runs the Main program in `A.dfy` with no command-line arguments
+  - `dafny run A.dfy --no-verify` -- builds the Main program in `A.dfy` using the `--no-verify` option, and then runs the program with no command-line arguments
+  - `dafny run A.dfy 1 2 3 B.dfy` -- builds the Main program in `A.dfy` and
+then runs it with the four command-line arguments `1 2 3 B.dfy`
+  - `dafny run A.dfy 1 2 3 --input B.dfy` -- builds the Main program in `A.dfy` and `B.dfy`, and
```

Comment:
I'm a bit concerned by this one, I need confirmation of @keyboardDrummer

Reply:
Looks correct to me